listener.go

```diff
@@ -69,7 +69,7 @@ func (l *listener) process() {
 	for {
 		conn, err := l.wrapped.Accept()
 		if err != nil {
-			l.errCh <- &netError{err.Error(), false, false}
+			l.errCh <- err
```

Comment:
We just return the original error. That way, the listener can detect whether it's a temporary network error and retry.